Starting with 2.14 the GTK+ FileSystemError is no longer available due
to the move to GIO internally. This structure was also never intended
to be exposed, so we are removing it here. See bug
http://bugs.gentoo.org/show_bug.cgi?id=238837 for more details.

--- gtk/src/rbgtkfilechooser.c.orig	2006-12-29 14:17:28.000000000 +0100
+++ gtk/src/rbgtkfilechooser.c	2008-09-28 09:40:10.573694830 +0200
@@ -16,10 +16,6 @@
 
 #define _SELF(self) GTK_FILE_CHOOSER(RVAL2GOBJ(self))
 
-/* For error handling */
-#define GTK_FILE_SYSTEM_ENABLE_UNSUPPORTED
-#include <gtk/gtkfilesystem.h>
-
 static VALUE
 gslist2ary_free(list)
     GSList* list;
@@ -367,14 +363,6 @@
     /* GtkFileChooserError */
     G_DEF_ERROR(GTK_FILE_CHOOSER_ERROR, "FileChooserError", mGtk, rb_eRuntimeError, 
                 GTK_TYPE_FILE_CHOOSER_ERROR);
-    /* GtkFileSystemError */
-    fse = G_DEF_ERROR2(GTK_FILE_SYSTEM_ERROR, "FileSystemError", mGtk, rb_eRuntimeError);
-    rb_define_const(fse, "NONEXISTENT", INT2NUM(GTK_FILE_SYSTEM_ERROR_NONEXISTENT));
-    rb_define_const(fse, "NOT_FOLDER", INT2NUM(GTK_FILE_SYSTEM_ERROR_NOT_FOLDER));
-    rb_define_const(fse, "INVALID_URI", INT2NUM(GTK_FILE_SYSTEM_ERROR_INVALID_URI));
-    rb_define_const(fse, "BAD_FILENAME", INT2NUM(GTK_FILE_SYSTEM_ERROR_BAD_FILENAME));
-    rb_define_const(fse, "FAILED", INT2NUM(GTK_FILE_SYSTEM_ERROR_FAILED));
-    rb_define_const(fse, "ALREADY_EXSITS", INT2NUM(GTK_FILE_SYSTEM_ERROR_ALREADY_EXISTS));
 
 #if GTK_CHECK_VERSION(2,8,0)
     /* GtkFileChooserConfirmation */
